move_base/src/move_base.cpp

```diff
@@ -1200,4 +1207,19 @@ namespace move_base {
 
     return true;
   }
+
+  // publish in context of action feedback, path left and time left to the goal
+  void MoveBase::publishPathMsg(const geometry_msgs::PoseStamped& current_position, const geometry_msgs::PoseStamped& goal)
+  {
+    move_base::Path msg;
+    msg.current_pose = current_position;
+    msg.goal_pose = goal;
+    msg.dist_to_goal = distance(current_position, goal);
```

Comment:
Currently, you compute straight line distance instead of global plan length left.

Reply:
I know, but I can call planner in context of action

Reply:
You don't need to compute it, it's already precomputed (look for `latest_plan_`, `controller_plan_`)

Reply:
Thanks